src/nvim/api/vim.c

```diff
@@ -2687,6 +2687,9 @@ void nvim__screenshot(String path)
 
 static void clear_provider(DecorProvider *p)
 {
+  if (p == NULL) {
```

Comment:
hmm this should rather have the `NONNULL` attribute (I can fix it next time I touch this function which will be very soon).